src/Warehouse.java

```diff
@@ -60,12 +64,32 @@ public List<Product> getProductsInStock() {
     public boolean isEmpty() {
         return devices.isEmpty();
     }
+    public double calculateAveragePurchasePriceByDeviceType(Devices deviceType) throws InvalidSearchFilterException {
```

Comment:
Attento questo metodo è finito in questo commit. Inoltre è la versione errata ancora. Probabilmente hai staccato un branch non dal main ma dal branch in cui già stavi lavorando.